dockers/docker-orchagent/orchagent.sh

```diff
@@ -73,9 +73,9 @@ else
 fi
 
 # Enable ZMQ for SmartSwitch
-LOCALHOST_SUBTYPE=`sonic-db-cli CONFIG_DB hget localhost "subtype"`
+LOCALHOST_SUBTYPE=`sonic-db-cli CONFIG_DB hget "DEVICE_METADATA|localhost" "subtype"`
 if [[ x"${LOCALHOST_SUBTYPE}" == x"SmartSwitch" ]]; then
-    ORCHAGENT_ARGS+=" -q tcp://127.0.0.1:8100"
+    ORCHAGENT_ARGS+=" -q tcp://0.0.0.0:8100"
```

Comment:
I feel listening 0.0.0.0 in the production environment isn't a good idea. Here is a sample code that I did in the database container to fetch the midplane network address you can refer.

https://github.com/sonic-net/sonic-buildimage/pull/17562/files#diff-8b146f363257f0e5092d24d8eda725a077aa65d60814c58f7be097cc88f8c07cR33-R41

Reply:
Thanks, fixed, changed to listen eth0 IPV4 address.

Reply:
But DPU does not use eth0, it should use tmfifo port.

Reply:
It would be named eth0-midplane

Reply:
Is this address allocated by dhcp?

Reply:
yes

Reply:
Can we guarantee that this is address is allocated before orchagent.sh?
If not, we can't set this address in orchagent.sh.

Reply:
I think this is a common issue for both database/orch and maybe other service. It should not handle by orchagent or database container by monitinor interface change.

The platform need guarantee interface address ready before services start:
1. Add a service to check the interface ready and make database/orch depends on that service.
2. eth0-midplane shoud be part of minigraph override, after HDCP config regerate on host, reload minigraph on all DPU.
3. Add code in orchagent start script to wait untill interface ready.